broker/src/main/java/io/camunda/zeebe/broker/partitioning/topology/TopologyPartitionListenerImpl.java

```diff
@@ -7,24 +7,25 @@
  */
 package io.camunda.zeebe.broker.partitioning.topology;
 
+import io.camunda.zeebe.engine.api.ProcessingScheduleService;
 import io.camunda.zeebe.protocol.impl.encoding.BrokerInfo;
 import io.camunda.zeebe.protocol.record.PartitionRole;
-import io.camunda.zeebe.scheduler.ActorControl;
+import java.time.Duration;
 import org.agrona.collections.Int2IntHashMap;
 
 public final class TopologyPartitionListenerImpl implements TopologyPartitionListener {
 
   private final Int2IntHashMap partitionLeaders = new Int2IntHashMap(-1);
-  private final ActorControl actor;
+  private final ProcessingScheduleService scheduleService;
 
-  public TopologyPartitionListenerImpl(final ActorControl actor) {
-    this.actor = actor;
+  public TopologyPartitionListenerImpl(final ProcessingScheduleService scheduleService) {
+    this.scheduleService = scheduleService;
```

Comment:
:question: this is a bit confusing to me. Why we use the ProcessingScheduleService here? This has nothing to do with the processing ?

Reply:
This change came about, because historically `TopologyPartitionListenerImpl` was using the actor retrieved from `RecordProcessorContext` (see `PartitionFactory.createFactory(...)`). Maybe it is important they use the same actor???

But I agree that this seems like an unwanted dependency.

Reply:
Also, this is the only place (except for tests) where `TypedRecordProcessorContext.getScheduleService(...)` is called. So if we could use another actor here, we could also slim down `TypedRecordProcessorContext`